tests/cuistl/test_cusparsematrix.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+  Copyright 2022-2023 SINTEF AS
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <config.h>
+
+#define BOOST_TEST_MODULE TestCuSparseMatrix
+
+#include <boost/test/unit_test.hpp>
+#include <dune/istl/bcrsmatrix.hh>
+#include <memory>
+#include <opm/simulators/linalg/cuistl/CuSparseMatrix.hpp>
+#include <opm/simulators/linalg/cuistl/CuVector.hpp>
+#include <opm/simulators/linalg/cuistl/detail/cuda_safe_call.hpp>
+#include <random>
+
+BOOST_AUTO_TEST_CASE(TestConstruction1D)
+{
+    // Here we will test a simple 1D finite difference scheme for
+    // the Laplace equation:
+    //
+    //    -\Delta u = f on [0,1]
+    //
+    // Using a central difference approximation of \Delta u, this can
+    // be approximated by
+    //
+    //    -(u_{i+1}-2u_i+u_{i-1})/Dx^2 = f(x_i)
+    //
+    // giving rise to the matrix
+    //
+    //     -2  1  0  0 ... 0  0
+    //      1 -2  1  0  0 ... 0
+    //      ....
+    //      0  0  0  ...1 -2  1
+    //      0  0  0  ...   1 -2
+
+    const int N = 5;
+    const int nonZeroes = N * 3 - 2;
+    using M = Dune::FieldMatrix<double, 1, 1>;
```

Comment:
This seems to contradict a statement elsewhere that block sizes must be > 1. Seems to me that other statement is wrong, and we indeed support block size 1?

Reply:
This only tests construction, where we don't have any limitations. Only arithmetic operations  (mv etc) require block size 1.